hooks/generatePdf.js

```diff
@@ -0,0 +1,28 @@
+const puppeteer = require('puppeteer');
+const path = require('path');
+
+module.exports = {
+    'generate:after': generatePdf
+};
+
+async function generatePdf(generator) {
+    const targetDir = generator.targetDir;
+    const parameters = generator.templateParams;
+    
+    //all actions of this hook depend on parameters passed by the user, if non are provided we should just stop the hook
+    if (!parameters || parameters.pdf !== 'true') return;
+
+    const browser = await puppeteer.launch();
+    const page = await browser.newPage();
+
+    await page.goto(`file://${path.join(targetDir, 'index.html')}`, { waitUntil: 'networkidle0' });
+
+    await page.$$eval('div.payload-examples > div.js-prop', links => links.forEach(link => link.click()));
+    await page.$$eval('div.headers-examples > div.js-prop', links => links.forEach(link => link.click()));
+    await page.$$eval('div.message-examples > div.js-prop', links => links.forEach(link => link.click()));
```

Comment:
```suggestion
    await page.$$eval('div.all-messages div.js-prop', links => links.forEach(link => link.click()));
```

`>` in the selector is used to point to the element that is a direct child so in the examples section, it worked. In the case of messages, `div.js-prop` is a nested child, so just `>` must be removed and  selector will pick all divs, no matter how deep they are

Reply:
Oh! I misunderstood the meaning of `>` Thanks for the clarification.